src/main/groovy/net/minecraftforge/gradleutils/GradleUtilsExtension.groovy

```diff
@@ -5,138 +5,217 @@
+import net.minecraftforge.gradleutils.gitversion.GitVersionExtension
+import org.gradle.TaskExecutionRequest
 import org.gradle.api.Project
-import org.gradle.api.file.Directory
+import org.gradle.api.Task
 import org.gradle.api.file.DirectoryProperty
-import org.gradle.api.provider.Provider
+import org.gradle.api.problems.ProblemId
+import org.gradle.api.problems.Problems
+import org.gradle.api.tasks.TaskProvider
 
 import javax.inject.Inject
 
 @CompileStatic
 class GradleUtilsExtension {
+    static final String NAME = 'gradleutils'
+
     private final Project project
-    final DirectoryProperty gitRoot
-    private final Provider<Map<String, String>> gitInfo
+    @PackageScope final Problems problems
+
+    final PomUtils pom
+
+    @Deprecated(forRemoval = true, since = '2.4') final DirectoryProperty gitRoot
+    @Deprecated(forRemoval = true, since = '2.4') @Lazy Map<String, String> gitInfo = {
+        var id = ProblemId.create('deprecated-git-info', 'Usage of deprecated Git Info', GradleUtils.PROBLEM_GROUP)
+        this.problems.reporter.report(id, spec ->
+            spec.contextualLabel('GradleUtils 3.0 Deprecations')
+                .details("""
+                        This project is still using 'gradleutils.gitInfo'.
+                        It has been deprecated and will be removed in GradleUtils 3.0.""")
+                .solution("Consider using 'gradleutils.version.info' instead.")
+        )
+
+        var version = this.project.extensions.getByType(GitVersionExtension).version
+        [
+            dir          : version.gitDir.absolutePath,
+            tag          : version.info.tag,
+            offset       : version.info.offset,
+            hash         : version.info.hash,
+            branch       : version.info.branch,
+            commit       : version.info.commit,
+            abbreviatedId: version.info.abbreviatedId,
+            url          : version.info.url
+        ].tap { it.removeAll { it.value == null } } as Map<String, String>
+    }()
 
     @Inject
-    GradleUtilsExtension(Project project) {
+    GradleUtilsExtension(Project project, Problems problems) {
         this.project = project
+        this.problems = problems
+        this.pom = new PomUtils(problems, project.extensions.getByType(GitVersionExtension))
+
+        this.setupTasks()
+
+        this.gitRoot = project.objects.directoryProperty().convention project.layout.projectDirectory
+    }
+
+    private void setupTasks() {
+        // Actions Workflow Generator
+        GenerateActionsWorkflow.register(this.project)
 
-        this.gitRoot = project.objects.directoryProperty().convention(project.layout.projectDirectory)
-        this.gitInfo = project.objects.mapProperty(String, String)
-                .convention(gitRoot.map((Directory dir) -> GradleUtils.gitInfo(dir.asFile)))
+        // Legacy CI Tasks
+        GradleUtils.setupCITasks(this.project)
+    }
+
+    void requestTask(TaskProvider<? extends Task> task) {
```

Comment:
What is this for?

Reply:
Unused garbage that allows a task to be guaranteed to run at project sync. No need for it anymore, I'll yeet it.